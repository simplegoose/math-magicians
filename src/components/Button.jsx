import React from 'react';
import PropTypes from 'prop-types';

export default function Button({ className, onClick, children }) {
  return (
    <button type="button" className={className} onClick={() => onClick()}>{children}</button>
  );
}

Button.propTypes = {
  className: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};
