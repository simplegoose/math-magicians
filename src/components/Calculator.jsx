import React, { useState } from 'react';
import calculate from '../logic/calculate';
import Button from './Button';

export default function Calculator() {
  const [data, setData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const doCalculation = (btnName) => {
    const obj = calculate(data, btnName);

    setData(obj);
  };

  const getValue = (data) => {
    if (data.next !== null) return data.next;

    if (data.total !== null) return data.total;

    return 0;
  };

  return (
    <main>
      <section className="calc-section">
        <div className="calculator">
          <div className="screen">{getValue(data)}</div>
          <Button type="button" className="btn" onClick={() => doCalculation('AC')}>AC</Button>
          <Button type="button" className="btn" onClick={() => doCalculation('+/-')}>+/-</Button>
          <Button type="button" className="btn" onClick={() => doCalculation('%')}>%</Button>
          <Button type="button" className="btn bg-orange" onClick={() => doCalculation('รท')}>&divide;</Button>
          <Button type="button" className="btn" onClick={() => doCalculation('7')}>7</Button>
          <Button type="button" className="btn" onClick={() => doCalculation('8')}>8</Button>
          <Button type="button" className="btn" onClick={() => doCalculation('9')}>9</Button>
          <Button type="button" className="btn bg-orange" onClick={() => doCalculation('x')}>x</Button>
          <Button type="button" className="btn" onClick={() => doCalculation('4')}>4</Button>
          <Button type="button" className="btn" onClick={() => doCalculation('5')}>5</Button>
          <Button type="button" className="btn" onClick={() => doCalculation('6')}>6</Button>
          <Button type="button" className="btn bg-orange" onClick={() => doCalculation('-')}>-</Button>
          <Button type="button" className="btn" onClick={() => doCalculation('1')}>1</Button>
          <Button type="button" className="btn" onClick={() => doCalculation('2')}>2</Button>
          <Button type="button" className="btn" onClick={() => doCalculation('3')}>3</Button>
          <Button type="button" className="btn bg-orange" onClick={() => doCalculation('+')}>+</Button>
          <Button type="button" className="btn span-2" onClick={() => doCalculation('0')}>0</Button>
          <Button type="button" className="btn" onClick={() => doCalculation('.')}>.</Button>
          <Button type="button" className="btn bg-orange" onClick={() => doCalculation('=')}>=</Button>
        </div>
      </section>
    </main>
  );
}
