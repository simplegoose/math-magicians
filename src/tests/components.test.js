import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../routes/Quote';
import Calculator from '../routes/Calculator';
import CalculatorComponent from '../components/Calculator'
import Button from '../components/Button';
import Layout from '../components/Layout'
import Navbar from '../components/Navbar';
import { BrowserRouter } from 'react-router-dom';

it('Quotes renders correctly', () => {
    const tree = renderer
        .create(<Quote />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Calculator route renders correctly', () => {
    const tree = renderer
        .create(<Calculator />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Calculator component renders correctly', () => {
    const tree = renderer
        .create(<CalculatorComponent />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Button component renders correctly', () => {
    const tree = renderer
        .create(<Button />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Layout component renders correctly', () => {
    const tree = renderer
        .create(
        <BrowserRouter><Layout /></BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Navbar component renders correctly', () => {
    const tree = renderer
        .create(
        <BrowserRouter><Navbar /></BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

