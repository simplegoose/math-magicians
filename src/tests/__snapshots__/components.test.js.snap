// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<button
  onClick={[Function]}
  type="button"
/>
`;

exports[`Calculator component renders correctly 1`] = `
<section
  className="calc-section"
>
  <div
    className="calculator"
  >
    <div
      className="screen"
    >
      0
    </div>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn bg-orange"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn bg-orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn bg-orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn bg-orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="btn span-2"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn bg-orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</section>
`;

exports[`Calculator route renders correctly 1`] = `
<main
  className="calc_main"
>
  <div
    className="calc_title"
  >
    <h2>
      Let’s do some math!
    </h2>
  </div>
  <section
    className="calc-section"
  >
    <div
      className="calculator"
    >
      <div
        className="screen"
      >
        0
      </div>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn bg-orange"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn bg-orange"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn bg-orange"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn bg-orange"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn span-2"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn bg-orange"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </section>
</main>
`;

exports[`Layout component renders correctly 1`] = `
<div>
  <nav
    className="nav"
  >
    <h1
      className="title"
    >
      Math Magicians
    </h1>
    <ul
      className="nav_buttons_container"
    >
      <li
        className="nav_buttons"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav_buttons"
      >
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="nav_buttons"
      >
        <a
          className=""
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Navbar component renders correctly 1`] = `
<nav
  className="nav"
>
  <h1
    className="title"
  >
    Math Magicians
  </h1>
  <ul
    className="nav_buttons_container"
  >
    <li
      className="nav_buttons"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav_buttons"
    >
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="nav_buttons"
    >
      <a
        className=""
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Quotes renders correctly 1`] = `
<main
  className="quote_main"
>
  <div
    className="quote"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit.  Autem, labore ratione assumenda doloremque nobis tempore nesciunt,  aperiam, in omnis fuga exercitationem? Maiores quia delectus quidem  sequi aliquid voluptas excepturi laudantium?
  </div>
</main>
`;
